solver/src/settlement.rs

```diff
@@ -204,99 +251,249 @@ mod tests {
         assert!(settlement.encode_trades().is_some());
     }
 
+    // Helper function to save some repeatition below.
+    fn r(u: u128) -> BigRational {
```

Comment:
Nit: I guess this has been reduced to just r because calling it rat would also lead the reader to have to check what it is anyway and rationalize is far too long. If not, I might suggest calling it rational or at the very least rat. I know it's mathematically sound and much more concise as is, but I'm just hoping to prevent the reader from actually having to jump to the definition.

Reply:
I think calling it "rat" could tranform

https://github.com/gnosis/gp-v2-services/blob/79b12404652c7467002858ff079aeb0b5d8924dd/solver/src/settlement.rs#L406-L447

into

![tenor](https://user-images.githubusercontent.com/624308/111584736-2933f280-87b6-11eb-8119-8aaf7eef0a9d.gif)

:)

(now seriously, @vkgnosis suggested the name, but I think one letter is already too many letters - there should be an implicit conversion in this case)

Reply:
> hoping to prevent the reader from actually having to jump to the definition

The function is one line long and just above the tests so my personal preference is as short as possible because it makes the asserts more readable.